### PR TITLE
Mark conflict on rector >=0.12.18

### PR DESCRIPTION
**Pull request by mglaman**
## Description
This project's default config is not compatible with latest Rector changes in 0.12.21

## To Test
Install. Get rector 0.12.21, copy default config, get

```
 [ERROR] Undefined constant Rector\Core\Configuration\Option::IMPORT_DOC_BLOCKS
```

## Drupal.org issue

https://www.drupal.org/project/rector/issues/3278468

### CODE REVIEW

**Comment by neclimdul** [comment]
Main commit with releases https://github.com/rectorphp/rector/commit/071d7cb7880579b2279aeff3b377d81c94d0836c
Breaking commit in main repo https://github.com/rectorphp/rector-src/commit/c35cc97d381f97c602cccb86ccf1d688a869cbdd

**Comment by mglaman** [comment]
The one failure is on missing `setUp(): void` which should be fixed by our config.

File: composer.json [hunk 1]
```diff
@@ -68,7 +71,14 @@
...
         "phpunit/phpunit": "^9.5",
-        "rector/rector-src": "dev-main",
+        "rector/rector-src": "dev-main#77a4582d5044446b8da43ab91a09082fa0413a23",
+        "rector/rector-cakephp": "dev-main#eab7b7992e3462c40b2eb8f7ab6d2013eaf20c76",
+        "rector/rector-doctrine": "dev-main#21431cc9e81990191be9c235e67c631ed518d0a3",
+        "rector/rector-laravel": "dev-main#31d9892d1f2abd41a118589ad76a065bb24150ea",
+        "rector/rector-nette": "dev-main#7aa632f0fd210b0795cb92b98e648164049db80e",
+        "rector/rector-phpoffice": "dev-main#1e12437861bb9e06f776b9f9be5811a7bf219729",
+        "rector/rector-phpunit": "dev-main#baf6f9eaa1eadc03d61ac6f414077d85b0540dfa",
+        "rector/rector-symfony": "dev-main#a3ad2a20c028535c7d26ba85bed729d838eb9297",
```
Comment (mglaman) [inline composer.json:81]: So. `rector/rector-src` is required to run tests. It has requirements on _all_ of these at `dev-main`, which include the breaking config change. These are the commits _prior to_ the configuration change.

This allows us to fix the blocking issues and slowly bump back to `rector/rector-src:dev-main`

File: config/drupal-8/drupal-8-all-deprecations.php [hunk 2]
```diff
@@ -11,7 +11,6 @@
 
     $containerConfigurator->import(PHPUnitSetList::PHPUNIT_60);
     $containerConfigurator->import(PHPUnitSetList::PHPUNIT_70);
-    $containerConfigurator->import(PHPUnitSetList::PHPUNIT_75);
```
Comment (mglaman) [inline config/drupal-8/drupal-8-all-deprecations.php]: I should revert this and we fix it directly in https://www.drupal.org/project/rector/issues/3277704

File: composer.json [hunk 3]
```diff
@@ -13,6 +13,9 @@
         "rector/rector": "~0.12",
         "webflo/drupal-finder": "^1.2"
     },
+    "conflict": {
+        "rector/rector": ">0.12.17"
+    },
```
Comment (mglaman) [inline composer.json:18]: next follow up is to bump this to 0.12.18 and fix https://www.drupal.org/project/rector/issues/3271102

**Comment by mglaman** [comment]
The functional test is failing with auto import as changed in https://github.com/palantirnet/drupal-rector/pull/194, but is failing due to other issues.

File: rector.php [hunk 4]
```diff
@@ -28,6 +28,7 @@
     $parameters->set(Option::FILE_EXTENSIONS, ['php', 'module', 'theme', 'install', 'profile', 'inc', 'engine']);
     $parameters->set(Option::AUTO_IMPORT_NAMES, true);
     $parameters->set(Option::IMPORT_SHORT_CLASSES, false);
+    $parameters->set(Option::APPLY_AUTO_IMPORT_NAMES_ON_CHANGED_FILES_ONLY, true);
     // @phpstan-ignore-next-line
     $parameters->set(Option::IMPORT_DOC_BLOCKS, false);
```
Comment (mglaman) [inline rector.php]: See https://github.com/rectorphp/rector-src/pull/1817/files#diff-f3a81d2f0188eec529c7c462d49878ed6d8cdabe7090818a5496c493aa80ba5b

**Pull request merged**